src/Nethermind/Nethermind.Synchronization/SyncServer.cs

```diff
@@ -180,7 +180,8 @@ public void AddNewBlock(Block block, ISyncPeer nodeWhoSentTheBlock)
 
             bool isBlockBeforeTheSyncPivot = block.Number < _pivotNumber;
             bool isBlockOlderThanMaxReorgAllows = block.Number < (_blockTree.Head?.Number ?? 0) - Sync.MaxReorgLength;
-            bool isBlockTotalDifficultyLow = block.TotalDifficulty < _blockTree.BestSuggestedHeader.TotalDifficulty;
+            bool isBlockTotalDifficultyLow = block.TotalDifficulty < _blockTree.BestSuggestedHeader.TotalDifficulty
+                                             && (_specProvider.TerminalTotalDifficulty == null || block.TotalDifficulty < _specProvider.TerminalTotalDifficulty); // terminal blocks with lower TTD might be useful for smooth merge transition
```

Comment:
What if there are two terminal block and the lower one appear later? I assume it must go through NP?

Reply:
If terminal block appears later that shouldn't be block choosen by CL clients to build chain on top of it. However, in this case we will get NP(childOfTerminalBlock) we will return SYNCING and we should be able to catch up with MergeBlockDownloader